app/Http/Controllers/GalaxyController.php

```diff
@@ -399,7 +399,7 @@ public function ajax(Request $request, PlayerService $player, PlanetServiceFacto
                 'playerId' => $player->getId(),
                 'settingsProbeCount' => 3,
                 'showOutlawWarning' => true,
-                'slotsColonized' => 3,
+                'slotsColonized' => $player->planets->count(),
```

Comment:
The `slotsColonized` should reflect the amount of planets that are colonized in the current system. The value you are using now is the amount of planets the current player has. This results in an incorrect count. 

See e.g. this screenshot. I have two planets, but the system I'm looking at has three planets colonized. So I expect it to show `3 Planets colonized` instead of the two that I see now.

<img width="803" alt="Screenshot 2024-10-16 at 15 15 26" src="https://github.com/user-attachments/assets/8eb569d8-163a-4254-ad89-4629b21b3757">

I think the output of the existing method `GalaxyController::getGalaxyArray()` could be used to check how many planets in the system are actually colonized.

Reply:
Hi @lanedirt,

I apologize for the misunderstanding regarding the count of colonized planets. I initially thought the count referred only to the player's planets. Now that you've clarified, I see that we should only count the colonized planets within the current system.

Here is the logic I've implemented to count the colonized planets:

```PHP
public function ajax(Request $request, PlayerService $player, PlanetServiceFactory $planetServiceFactory): JsonResponse
{
   // ...
    $galaxyContent = $this->getGalaxyArray($galaxy, $system, $player, $planetServiceFactory);
    $slotsColonized = 0;

    foreach ($galaxyContent as $galaxySystem) {
        if (isset($galaxySystem['planets'])) {
            $activePlanets = array_filter($galaxySystem['planets'], function ($planet) {
                return !isset($planet['isDestroyed']) || !$planet['isDestroyed'];
            });
            $slotsColonized += count($activePlanets);                
        }
    }
   // ...
}
```
I believe this implementation accurately counts the colonized planets in the current system. If you have any further thoughts on this or if you think the logic should be placed elsewhere, such as a different method or a specific pattern followed in the project, I’m open to any suggestions for improvement.

Thank you very much for your attention and time.

Reply:
Hi @gabrieltsants,

Thanks for tweaking the code. However, it's a bit difficult for me to review the changes based only on the snippet you provided. Could you push the changes in a new commit and add it to this PR so I can test it locally?

I'm happy to test the new version and provide feedback.

Reply:
Hi @lanedirt,

I’ve just committed the changes with the refactor you requested. You can now review the updates directly in the PR. Let me know if anything else is needed or if you encounter any issues while testing!

Looking forward to your feedback.